internal/notifications/notifications.go

```diff
@@ -0,0 +1,76 @@
+// Notifications that `cow.Containers` can raise and Notify clients about
+//
+// generalized from `internal/gcs/protocol.go` and `internal/guest/prot/protocol.go`
+package notifications
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/queue"
+)
+
+type NotificationType string
+
+type Notification interface {
```

Comment:
Maybe we should call this `Message`? Would make it easier for the rest of the code to refer to as `notifications.Message` instead of `notifications.Notification`

Reply:
I picked `Notification` because `Event` seemed heavily associated with containerd's events system, but I think between `Notification` (which the bridge uses),  `Event` (which mostly is just for interfacing with containerd), and `Message`, we should settle on a term that makes the most sense throughout